### PR TITLE
[feat] Update graphql schemas and small improvements

### PR DESCRIPTION
**Pull request by SammyBaek**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #102
* #98
* #97
* #93
* __->__ #92

## Changes made
- Update schema.graphql to include more types
- Fetch lazy references more efficiently
- Allow family lookup via LazyReference

## Testing
- [x] End-to-End

### CODE REVIEW

File: server/src/gql/resolver.py [hunk 1]
```diff
@@ -25,6 +26,20 @@ def serialize_oid(id: ObjectId) -> str:
     return str(id)
 
 
+date_scalar = ScalarType("Date")
+
+
+@date_scalar.serializer
+def serialize_date(date: datetime) -> str:
```
Comment (SammyBaek) [inline server/src/gql/resolver.py:33]: Otherwise, it returns as english i.e. `Wed, Nov...`

File: server/src/model/family.py [hunk 2]
```diff
@@ -22,16 +23,17 @@ class Family(Document):
     @property
     def mates(self) -> List[Mate]:
         """Fetch object for mate_ids."""
-        return [mate.fetch() for mate in self.mate_ids]
+        return list(Mate.objects.in_bulk([ref.id for ref in self.mate_ids]).values())
```
Comment (SammyBaek) [inline server/src/model/family.py:26]: Bulk fetching > fetching one-by-one:

Tried fetching 100 items
Screen Shot 2021-11-14 at 11 35 36 PM

File: server/src/model/family.py [hunk 3]
```diff
@@ -22,16 +23,17 @@ class Family(Document):
...
 
     def get_id(self) -> str:
         """Return string version of mongo oid."""
         return str(self.id)
 
     @staticmethod
     @error_bounded(None)
-    def lookup_family(id: str) -> Optional[Family]:
+    def lookup_family(family: Union[str, LazyReference]) -> Optional[Family]:
```
Comment (SammyBaek) [inline server/src/model/family.py:34]: Allow for `LazyReference`

File: server/tests/gql/test_resolver.py [hunk 4]
```diff
@@ -17,6 +18,16 @@ def test_serialize_oid() -> None:
     assert res == id_str
 
 
+def test_serialize_date() -> None:
+    """Test serialization from datetime to str."""
+    iso_str = "2021-11-17T23:40:37.347Z"  # new Date().toISOString() in JS
+    iso_res = "2021-11-17T23:40:37.347000Z"
+    date = datetime.fromisoformat(iso_str[:-1])
```
Comment (SammyBaek) [inline server/tests/gql/test_resolver.py:25]: python doesn't recognize `Z`

File: server/src/model/family.py (hunk 3 above)
Comment (EpicPi) [reply server/src/model/family.py:34]: This is painful to look at, but I guess it lets the syntax on the caller side make sense. I'm now questioning whether we should get rid of the `_id` on all the LazyRef fields.
Would be a diff ticket tho.

**Review by EpicPi — APPROVED** [review]
Great work - minor question  - we could probably do it in another ticket tho.

File: server/schema.graphql [hunk 5]
```diff
@@ -9,28 +9,43 @@ type Sync {
...
+  unassigned_family: Family!
+  families: [Family!]!
 }
 
 input NewMatesInput {
   name: String!
   lastSynced: Date
 }
 
+input NewAssignmentInput {
```
Comment (EpicPi) [inline server/schema.graphql:39]: This may better be called MateAssignmentInput?

**Review by EpicPi — CHANGES_REQUESTED** [review]
quick thing

File: server/schema.graphql [hunk 6]
```diff
@@ -9,28 +9,43 @@ type Sync {
...
+  mates: [Mate!]!
 }
 
-type Mate {
+type User {
   id: oid!
-  name: String!
-  sync_ids: [oid!]!
+  first_name: String!
+  picture_url: String
```
[the remark this answers is not in the document]
Comment (SammyBaek) [reply server/schema.graphql:28]: Yes, as a later ticket for sure. I'll add that in the board

File: server/src/model/family.py (hunk 3 above)
Comment (SammyBaek) [reply server/src/model/family.py:34]: unfortunately agreed

**Review by EpicPi — APPROVED** [review]
lgtm

**Pull request merged**